src/anim.ts

```diff
@@ -139,14 +139,8 @@ function step(state: State): void {
 
 function animate<A>(mutation: Mutation<A>, state: State): A {
   // clone state before mutating it
-  const prevPieces: cg.Pieces = {};
-  // clone pieces
-  for (let key in state.pieces) {
-    prevPieces[key] = {
-      role: state.pieces[key].role,
-      color: state.pieces[key].color
-    };
-  }
+  const prevPieces: cg.Pieces = {...state.pieces};
```

Comment:
Isn't this only referencing the same pieces differently? We need to do a real copy here, before we mutate the pieces. In order to diff the old and new pieces, and create the animation.